src/controller.cpp

```diff
@@ -930,21 +930,21 @@ std::string Controller::write_temporary_item(std::shared_ptr<RssItem> item)
 }
 
 void Controller::write_item(std::shared_ptr<RssItem> item,
-	const std::string& filename)
+	const Filepath& filename)
 {
 	const std::string save_path = cfg.get_configvalue("save-path");
-	auto spath = save_path.back() == '/' ? save_path : save_path + "/";
+	Filepath spath = save_path.back() == '/' ? save_path : save_path + "/";
 
-	std::string path;
-	switch (filename[0]) {
+	Filepath path;
+	switch (filename.to_locale_string()[0]) {
 	case '/':
-		path = filename;
+		path.push(filename);
 		break;
 	case '~':
 		path = utils::resolve_tilde(filename);
 		break;
 	default:
-		path = spath + filename;
+		path = spath.join(filename).clone();
 		break;
 	}
```

Comment:
Matching on the first byte of UTF-8 encoding is meh, although it should work fine. I see a couple options:

1. add this method to `Filepath` and rewrite the code to use it: https://doc.rust-lang.org/nightly/std/path/struct.Path.html#method.starts_with
2. add `is_absolute()` and `is_in_homedir()` methods to `Filepath` and rewrite the code with them. `is_absolute()` can defer to the same method on `PathBuf`, while `is_in_homedir()` you can implement on the Rust side with the aforementioned `starts_with()`.

Reply:
https://github.com/newsboat/newsboat/pull/2561/commits/34cc025b8d3e3180c4f26d12e63f651ac6515588